bundles/config/org.eclipse.smarthome.config.xml/src/main/java/org/eclipse/smarthome/config/xml/osgi/AbstractAsyncBundleProcessor.java

```diff
@@ -107,16 +107,16 @@ protected boolean isBundleRelevant(Bundle bundle) {
     }
 
     /**
-     * Checks for the existence of a given directors inside the bundle.
+     * Checks for the existence of a given directors inside the bundle and its attached fragments.
      *
      * Helper method which can be used in {@link #isBundleRelevant(Bundle)}.
      * 
      * @param bundle
      * @param path the directory name to look for
-     * @return <code>true</code> id the bundle contains the given directory
+     * @return <code>true</code> id the bundle or one of its attached fragments contain the given directory
      */
     protected final boolean isDirectoryPresent(Bundle bundle, String path) {
-        return bundle.getEntry(path) != null;
+        return bundle.findEntries(path, "*", true) != null;
```

Comment:
We do not know if the URL returned by getResource is a directory or a file. But the getEntry method does also not handle files or directories differently (or I do not get the point).
So using getReource seems good for me.

But perhaps we should rename the function to "isResourcePresent". Sure, this is not really part of your change, but IMHO using "directory" is not correct, as it returns also true for a file using that name (also before your PR).